### DISCUSSION TITLE
Multiple shipping addresses

### DISCUSSION THREAD

**Discussion by TLeszczynskiWA** [medusajs/medusa#4789] [Q&A]
Can MedusaJS be set up so that an Order owns multiple shipping addresses? I.e. one of the deliveries is shipped to customer's home and the other one is shipped to a pickup point.

**Comment by srindom** [comment]
This is a cool idea - think adding an address to Fulfillments would be a sensible change.

Could you explain the customer journey when a customer wants two shipments just to understand the requirements a bit further :)

**Reply by TLeszczynskiWA** [reply]
Sure! I think it'd be great if those could live at a line level.

Customer:
 1. adds SDA (small) and MDA (large) products to cart.
 2. proceeds to checkout.
 3. sets shipping method "Pick up in store" for SDA products - shipping address is the store's address (i.e. shop is next door and products will be available quickly)
 4. sets shipping method "Large package shipment"  for MDA products - shipping address is customer's home (i.e. to heavy to carry from store and may be available at a later point in time than SDA)
 5. completes checkout

The above example is the scenario that we have, however, it may also be that the customer makes a big order (i.e. real estate) to furnish new flats and wants products to be delivered to separate locations.

**Reply by cngann** [reply]
Here is a customer journey we use every day:
Customer views our product catalog, selecting multiple gifts for friends
Customer adds items to their cart
Each item is for a different recipient
Line items could be directed to different fulfillment providers such as USPS or FedEx

This is actually a pretty natural workflow for ecommerce

**Comment by radoslaw-sz** [comment]
hi @TLeszczynskiWA - do you still consider it as wanted feature?
I am looking for some idea which can be written as plugin for community, wondering if your issue is still open.
